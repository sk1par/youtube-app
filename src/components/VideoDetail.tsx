import Video from "../models/single-video.interface";

const VideoDetail = ({video}: Video) => {
  console.log(video);
  if (!video) {
    return <div>
       <h1>Enter search keyword to load...</h1>
       <br></br>
    </div>
  }

  const videoSrc = `https://www.youtube.com/embed/${video.id.videoId}`;
  return (
    <div>
      <div className="ui embed">
        <iframe src={videoSrc} allowFullScreen title="Video player" />
      </div>
      <div className="ui segment">
        <h4 className="ui header">{video.snippet.title}</h4>
        <p>{video.snippet.description}</p>
      </div>
    </div>
  );
};

export default VideoDetail;
