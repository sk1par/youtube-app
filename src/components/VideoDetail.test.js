import { useSelector } from 'react-redux';
import VideoDetail from './VideoDetail';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallowToJson } from 'enzyme-to-json';
import selectedVideo from '../actions/selectedVideoActions';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('react-redux', () => ({
  useDispatch: () => (fn) => () => fn(),
  useSelector: jest.fn(() => []),
}));

jest.mock('../actions/videosActions', () => {
  const selectedVideo = jest.fn();
  return selectedVideo;
});

const getVideo = () => ({
    id: {
      videoId: '1',
    },
    snippet: {
      description: 'test',
      title: 'test',
      thumbnails: {
        medium: {
          url: 'test',
        },
      },
    },
  });

it('should render search keyword div', () => {
    useSelector.mockReturnValueOnce([]);
    const wrapper = mount(<VideoDetail />);
    const searchDiv = wrapper.find('[data-auto-id="search-keyword-div"]');
    expect(searchDiv).toHaveLength(1);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('should render video detail div', () => {
    useSelector.mockReturnValueOnce(getVideo());
    const wrapper = shallow(<VideoDetail />);
    const videoDetailDiv = wrapper.find('[data-auto-id="video-details-div"]');
    expect(videoDetailDiv).toHaveLength(1);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});

