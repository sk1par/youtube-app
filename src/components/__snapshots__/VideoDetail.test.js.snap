// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render search keyword div 1`] = `
<VideoDetail>
  <NoData>
    <div
      data-auto-id="search-keyword-div"
    >
      <h1>
        Enter search keyword to load...
      </h1>
      <br />
    </div>
  </NoData>
</VideoDetail>
`;

exports[`should render video detail div 1`] = `
<div
  data-auto-id="video-details-div"
>
  <div
    className="ui embed"
  >
    <iframe
      allowFullScreen={true}
      src="https://www.youtube.com/embed/1"
      title="Video player"
    />
  </div>
  <div
    className="ui segment"
  >
    <h4
      className="ui header"
    >
      test
    </h4>
    <p>
      test
    </p>
  </div>
</div>
`;
